x-pack/plugin/sql/build.gradle

```diff
@@ -132,3 +132,11 @@ tasks.register("regen") {
     }
   }
 }
+
+task checkNoBwc {
+  dependsOn 'check'
```

Comment:
This wasn't exactly what I was thinking. What I was saying is that `checkNoBwc` should depend on `test`, `integTest`, `yamlRestTest`, etc. That way it runs those w/o running all the BWC stuff.